### PR TITLE
Feat/create interface to test network zappy gui

### PR DESCRIPTION
**Pull request by Thomaltarix**
I refactored the `Network` class.
In fact, to test our program correctly, a network interface is compulsory.
This interface will make us able to develop a simple server that just redirect a standard file descriptor as the server, and then just write into this file descriptor.

To do so, i made an `INetwork` interface, and an abstract class `ANetwork` to simplify all the classes that will inherit from.

To see more details, check this issue: #161

### CODE REVIEW

**Review by mathieurobert1 — COMMENTED** [review]
:+1:  Seems good

**Review by mathieurobert1 — DISMISSED** [review]

**Comment by theophile-jr** [comment]
Requested myself following Thoma's request.

**Review by theophile-jr — DISMISSED** [review]

File: gui/include/Engine/Engine.hpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ class Gui::Engine {
     private:
 
         ServerParser                _parser;        // Parser class for server's command.
-        std::shared_ptr<Network>    _network;       // Network class to connect to the server.
+        std::shared_ptr<INetwork>    _network;       // Network class to connect to the server.
```
Comment (mathieurobert1) [inline gui/include/Engine/Engine.hpp:54]:
```suggestion
        std::shared_ptr<INetwork>   _network;       // Network class to connect to the server.
```

File: gui/include/GUIUpdater/GUIUpdater.hpp [hunk 2]
```diff
@@ -52,7 +52,7 @@ class Gui::GUIUpdater {
     private:
 
         std::shared_ptr<GameData>       _gameData; // The GUI GameData to update.
-        std::shared_ptr<Network>        _network; // The network to send commands to the server.
+        std::shared_ptr<INetwork>        _network; // The network to send commands to the server.
```
Comment (mathieurobert1) [inline gui/include/GUIUpdater/GUIUpdater.hpp:55]:
```suggestion
        std::shared_ptr<INetwork>       _network; // The network to send commands to the server.
```

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some space lags in private classes arguments.

File: gui/include/Engine/Engine.hpp (hunk 1 above)
Comment (Thomaltarix) [reply gui/include/Engine/Engine.hpp]: Done

**Review by mathieurobert1 — APPROVED** [review]
Nice :+1:

**Review by theophile-jr — APPROVED** [review]
LGTM

**Pull request merged**